tensormap-client/src/components/DragAndDropCanvas/Canvas.js

```diff
@@ -0,0 +1,198 @@
+import React, { useState, useRef, useCallback,useMemo } from 'react';
+import { Grid,Form,Button } from 'semantic-ui-react';
+import ReactFlow, {
+  ReactFlowProvider,
+  addEdge,
+  useNodesState,
+  useEdgesState,
+  Controls,
+  Background,
+  BackgroundVariant
+} from 'reactflow';
+import 'reactflow/dist/style.css';
+import InputNode from './CustomNodes/InputNode/InputNode.js';
+import DenseNode from './CustomNodes/DenseNode/DenseNode.js';
+import FlattenNode from './CustomNodes/FlattenNode/FlattenNode.js';
+
+import Sidebar from './Sidebar.js';
+import PropertiesBar from '../PropertiesBar/PropertiesBar.js';
+import './Canvas.css';
+import { enableValidateButton,generateModelJSON } from './Helpers.js';
+import { validateModel } from '../../services/ModelServices';
+
+
+let id = 0;
+const getId = () => `dndnode_${id++}`;
+
+
+
+const Canvas = () => {
+  const reactFlowWrapper = useRef(null);
+  const [nodes, setNodes, onNodesChange] = useNodesState([]);
+  const [edges, setEdges, onEdgesChange] = useEdgesState([]);
+  const [reactFlowInstance, setReactFlowInstance] = useState(null);
+  const [formState,setFormState] = useState(
+    {
+      fileList:null,
+      totalDetails:null,
+      fieldsList: [{"text":"Select a file first", "value":null, "key":0,disabled:true}],
+      selectedFile:null,
+      targetField:null,
+      modalName:"",
+      problemType:null,
+      optimizer:null,
+      metric:null,
+      epochCount:0,
+      trainTestRatio:0,
+      disableButton:true,
+      modalOpen:false,
+      modelValidatedSuccessfully:false,
+  })
```

Comment:
Instead of writing this whole object in the useState call argument, can we define this `initial state` of state variable `formState` anywhere above? and later we can just pass the object variable name in the useState to make look code cleaner? What are your thoughts?

Reply:
Ack